engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java

```diff
@@ -233,6 +242,13 @@ public ExtendedSearchIntegrator getExtendedSearchIntegrator() {
 		return extendedIntegrator;
 	}
 
+	/**
+	 * Returns the names of the projection constants supported by a specific implementation in addition to projecting
+	 * actual field values. If a given projection name begins with {@link #HSEARCH_PROJECTION_FIELD_PREFIX} and is not
+	 * part of the set of constants returned by an implementation, an exception will be raised.
+	 */
+	protected abstract Set<String> getSupportedProjectionConstants();
```

Comment:
I'm using that style - providing contents for the "main" section, but omitting `@return` if it doesn't add any value - on purpose actually. It reads better that way in rendered JavaDoc than the other way around.

To avoid redundancy, I rarely use both. If I do it, then I have a general description in the main part (as the case here), and a description of specific values in the `@return` part (e.g. "@return null if bla bla bla, foo bar baz otherwise.").

It's not something we ever formally agreed on, just the way I came to think is providing most benefit.